File: matakuliah flutter/lib/matakuliah.dart
class Matakuliah {
  int id;
  String nama;
  String kode;
  int sks;

  Matakuliah(
      {required this.id,
      required this.nama,
      required this.kode,
      required this.sks});

  factory Matakuliah.fromJson(Map<String, dynamic> json) {
    return Matakuliah(
      id: json['id'],
      nama: json['nama'],
      kode: json['kode'],
      sks: json['sks'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nama': nama,
      'kode': kode,
      'sks': sks,
    };
  }
}

File: MataKuliah/src/main/java/com/mahasiswa/matakuliah/model/MataKuliah.java
package com.mahasiswa.matakuliah.model;


import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;


@Entity
public class MataKuliah {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String kode;
    private String nama;
    private Integer sks;

    public MataKuliah() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getNama() {
        return nama;
    }

    public void setNama(String nama) {
        this.nama = nama;
    }

    public String getKode() {
        return kode;
    }

    public void setKode(String kode) {
        this.kode = kode;
    }

    public Integer getSks() {
        return sks;
    }

    public void setSks(Integer sks) {
        this.sks = sks;
    }

}
